mujoco_warp/_src/constraint.py

```diff
@@ -292,6 +293,9 @@ def _efc_equality_joint(
 
   efc_worldid_out[efcid] = worldid
 
+  for i in range(nv):
```

Comment:
Not that I mind passing in `nv` but generally I wonder why we don't do this:

```py
for i in range(efc_J_out.shape[1]):
  efc_J_out[efcid, i] = 0.0
```

do you think patterns like this are good?